api/v1alpha1/edgedevice_types.go

```diff
@@ -64,10 +64,25 @@ type HardwareProfileConfiguration struct {
 
 // EdgeDeviceStatus defines the observed state of EdgeDevice
 type EdgeDeviceStatus struct {
-	Phase                     string      `json:"phase,omitempty"`
-	LastSeenTime              metav1.Time `json:"lastSeenTime,omitempty"`
-	LastSyncedResourceVersion string      `json:"lastSyncedResourceVersion,omitempty"`
-	Hardware                  *Hardware   `json:"hardware,omitempty"`
+	Phase                     string       `json:"phase,omitempty"`
+	LastSeenTime              metav1.Time  `json:"lastSeenTime,omitempty"`
+	LastSyncedResourceVersion string       `json:"lastSyncedResourceVersion,omitempty"`
+	Hardware                  *Hardware    `json:"hardware,omitempty"`
+	Deployments               []Deployment `json:"deployments,omitempty"`
+}
```

Comment:
shouldn't conditions be added as well, so the EdgeDevice can be watched ? (unless this is outside of the scope of this PR)

Reply:
I deliberately didn't implement any conditions to simplify the PoC. Deployment and Device state changes are reflected in `status.phase` and `status.deployments.phase`. 
IMO the conditions should be added at a later stage to provide detailed state information.